Dockerfile

```diff
@@ -54,12 +54,13 @@ RUN apt-get update \
 
 # Install Ruby 2.6.6, update RubyGems, and install Bundler
 ENV BUNDLE_SILENCE_ROOT_WARNING=1
-RUN apt-get install -y software-properties-common \
+RUN apt-get update && apt-get install -y software-properties-common \
   && apt-add-repository ppa:brightbox/ruby-ng \
   && apt-get update \
   && apt-get install -y ruby2.6 ruby2.6-dev \
-  && gem update --system 3.0.3 \
-  && gem install bundler -v 1.17.3 --no-document
+  && gem update --system 3.2.14 \
```

Comment:
Have dug through changes on rubygems for the version, requirement and specifications class. There are some changes to sorting of the requirements class: https://github.com/rubygems/rubygems/pull/2652/files

Test suite looks green so seems we don't rely on this here. I know we rely on it somewhat in api which prevented us from upgrading to ruby 2.7 but don't think this change will affect api as `gem` is installed from the `ruby2.7` docker image there.